rust/main/agents/validator/src/server/merkle_tree_insertions/list_merkle_tree_insertions.rs

```diff
@@ -0,0 +1,205 @@
+use axum::{
```

Comment:
is there a way to put more of this logic in hyperlane-base so we don't have as much of a copy-paste between here and the relayer?

Reply:
Yeah, I think we can probably squeeze more logic into hyperlane-base.
But I feel like its not super necessary because its just 2 copy-paste